### PR TITLE
GraphQL API:Adding user to organization

### PR DESCRIPTION
**Pull request by abdul-shajin**
GraphQL endpoint to add a new user to the organization.
accepts : email (required), name(optional).
If a user with given email already exists then add that user to the org.
If a user with that email does not exist then create a user with that email and then add that user to the org.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -19,10 +19,27 @@ def name
     "#{first_name} #{last_name}".squish
   end
 
+  def name=(name)
+    user = self
+    user.first_name, user.last_name = name.split(/\s+/, 2)
+  end
+
+  def extract_firstname_from_email
```
Comment (tejaswinichile) [inline app/models/user.rb]: Why do we need to extract this?

Comment (abdul-shajin) [reply app/models/user.rb]: because first_name in user model is mandatory and in graphql, the name field is optional

Comment (tejaswinichile) [reply app/models/user.rb]: Then I guess we should have same validation at graphQL side and model.
@neerajdotname Need your thoughts on this. Do you know why we are keeping these different validation at different level?

Comment (neerajdotname) [reply app/models/user.rb]: We need to fix the model. First name should not be required. If user does not have name then anywhere we show user name we should show `Anonymous`.

**Comment by tejaswinichile** [comment]
LGTM 👍

**Comment by tejaswinichile** [comment]
#312

**Pull request merged**